src/rotater.c

```diff
@@ -7,11 +7,17 @@ int oLookUp(int** o, char searchChar, int i) {
 }
 
 int jump(int bwtIndex, char jumpChar, int* c, int** o) {
-    //TODO
-    return 0;
+    return c[jumpChar] + oLookUp(o, jumpChar, bwtIndex);
 }
-struct Range* rotateString(char* string, int stringLen, int* c, int** o, char* bwt) {
-    //TODO
+
+struct Range* rotateString(char* string, int stringLen, int* c, int** o, char* bwt, int btwLen) {
     struct Range* r = malloc(sizeof(*r));
+    r->start = 0;
+    r->end = btwLen;
+    for(int i=0; i<stringLen; i++) {
+        if(r->start==r->end) return r;
+        //TODO jump
+    }
+
     return r;
 }
```

Comment:
I'm not sure what you are doing here, but any sub-step in these algorithms that iterate `for (int i=0; i < strLen; i++)` is highly suspicious.

You are aware, I hope, that you never have to rotate any strings in this algorithm. It is only something we *implicitly* do. Otherwise, the suffix array handles it for the construction of BWT, and the rank-select queries handle rotations when we search. At no point do we actually need to rotate a string.

Also, if you wanted rotations, using an offset and the modulus operator handles it efficiently in $O(1)$.

Reply:
This is search, where we rotate with all the chars in a string. 
In the version after this, where the todo is implemented, it should be doing it implicit with the rank-select queries, as you said.  
Can you comment on the newest version?

Reply:
I certainly plan to :)